### PR TITLE
Fix a few things and other QOL changes

### PR DESCRIPTION
**Pull request by Andre601**
Not a detailed or thought out PR... Just some stuff I fixed...
Initially wanted to add meta tags to the pages but now decided to just push this...

### CODE REVIEW

File: docs/contributing/structure-guide.md [hunk 1]
```diff
@@ -176,6 +184,10 @@ The `@return` for simple RestActions is as follows:
 ```
 
 ### Example Template
+
+!!! note inline end
```
Comment (Aikovdp) [inline docs/contributing/structure-guide.md:188]: I don't know if we should really use an inline admonition here? This is quite a big code block, and having an inline admonition there really reduces its size by a lot, making you scroll down to the bottom of the code block to use the horizontal scroll block. That's why I didn't inline it and just put it at the end.

**Review by Aikovdp — COMMENTED** [review]
This looks alright to me! 

The `<br>` tags were used everywhere in the original wiki, what do you think about replacing them with the double spaces @DV8FromTheWorld? It's nicer to read in unrendered markdown for sure, but it does make it less obvious when trying to edit it. Another option could to use `\` instead, to get rid of the html tag but still make it clear that there's a line break there.

File: docs/contributing/structure-guide.md (hunk 1 above)
Comment (Andre601) [reply docs/contributing/structure-guide.md:188]: Kinda agree. An alternative could be to use codeblock annotations to point out important sections.

**Comment by Andre601** [comment]
> This looks alright to me!
> 
> The `<br>` tags were used everywhere in the original wiki, what do you think about replacing them with the double spaces @DV8FromTheWorld? It's nicer to read in unrendered markdown for sure, but it does make it less obvious when trying to edit it. Another option could to use `\` instead, to get rid of the html tag but still make it clear that there's a line break there.

I for my part think that the wiki should be restructured in this area, since having stuff like a set max width makes no real sense when using (online) editors and only causes confusion.

If we use markdown, then we should utilize all its features as good as we can. After all is the purpose of markdown to simplify HTML markup and make it more readable...

So tl;dr I say we should get rid of the `<br>` tags and the overall width limitation.

File: docs/contributing/structure-guide.md (hunk 1 above)
Comment (Aikovdp) [reply docs/contributing/structure-guide.md:188]: Yeah, that'd be nice. They're not as obvious as the admonitions though, so I suppose that'd be dependent on how important the note is then

**Comment by Andre601** [comment]
I've pushed a change that switched out the inline admonition block with a code annotation.

Additionally did I move the image and SVG file into an `images` folder, since I also added a CSS file overriding the default code annotation colour to the primary one (The grey was barely visible on the code block with large comments)

**Comment by DV8FromTheWorld** [comment]
Wait, do you mean 2 spaces at the end of the line to indicate a `<br>`? I don't know why, but I thought you meant 2 newlines in the docs.

I definitely _don't_ want to use 2 trailing spaces to indicate a `<br>` as editors will remove trailing spaces during optimization workflows (Intellij runs this by default on file unfocus)

**Comment by Andre601** [comment]
> I definitely _don't_ want to use 2 trailing spaces to indicate a `<br>` as editors will remove trailing spaces during optimization workflows (Intellij runs this by default on file unfocus).

Idk what configuration you actually use, but IJ never did this in my case. The files always kept their trailing spaces in markdown files.

I'm generally not a fan of using HTML stuff inside markdown when there is a MD variant. As stupid as it may be.
Having a `<br>` in front of every new line for a simple line break makes sense for Javadocs, but for Markdown is it imo just bad.

Will re-add the `<br>`'s tho.

File: docs/contributing/structure-guide.md (hunk 1 above)
Comment (Andre601) [reply docs/contributing/structure-guide.md]: I did switch it with a code annotation, which imo looks decent enough and not too massive here.

**Comment by Andre601** [comment]
I've reverted the line breaks to using the `<br>` tags again.

One thing I would still ask for is, to get rid of this nonsense per-line limit.
Having lines break at a certain limit is just annoying because you can't really see if there is a new line or not and in terms of GitHub and the website itself does it also not matter as in both cases you don't see it by default (unless you look at the plain text in the markdown file).

Also, Editors will either auto-wrap it (Like GitHub) or just expand to the left which isn't even that bad.

If you want to keep this, then I'm sure are there some GH Actions and tools that could auto-format markdown to keep this.

File: docs/contributing/contributing.md [hunk 2]
```diff
@@ -57,10 +62,9 @@ Based on the current `upstream/development` (or `upstream/docs/development`) cha
         - Follow the [Structure Guide](structure-guide.md)
         - Compare your code style to the one used all over JDA and ensure you
           do not break the consistency (if you find issues in the JDA style you can include and update it)
-
+
```
Comment (Aikovdp) [inline docs/contributing/contributing.md:65]: Was this added for a reason? or was that just an oopsie

File: docs/contributing/structure-guide.md [hunk 3]
```diff
@@ -163,11 +170,12 @@ you have to document possible `ErrorResponses` that can follow from this request
  * </ul>
  */
 ```
+
 !!! note
     Always make these your last description paragraph
     <br>Taken from: [MessageChannel](https://github.com/DV8FromTheWorld/JDA/blob/master/src/main/java/net/dv8tion/jda/core/entities/MessageChannel.java#L169-L179)
 
-The `@return` for simple RestActions is as follows:
+The `@return` for simple RestActions is as follows:
```
Comment (Aikovdp) [inline docs/contributing/structure-guide.md:178]: You've missed a double space here

**Review by Aikovdp — CHANGES_REQUESTED** [review]
Would you be able to move `colors.css` to `stylesheets/` instead of `assets/css/`, and move the logos back to their original location? I know it doesn't technically make a difference, but it's closer to how the theme documents it too https://squidfunk.github.io/mkdocs-material/customization/, so that would make it easier for collaborators to figure out the structure, and that makes it worth it to me.

Everything else looks good to me!

File: docs/contributing/contributing.md (hunk 2 above)
Comment (Andre601) [reply docs/contributing/contributing.md:65]: Probs GitHub Desktop messing around

File: docs/contributing/structure-guide.md [hunk 4]
```diff
@@ -163,19 +170,21 @@ you have to document possible `ErrorResponses` that can follow from this request
...
  */
 ```
+
 !!! note
     Always make these your last description paragraph
     <br>Taken from: [MessageChannel](https://github.com/DV8FromTheWorld/JDA/blob/master/src/main/java/net/dv8tion/jda/core/entities/MessageChannel.java#L169-L179)
 
 The `@return` for simple RestActions is as follows:
-```java
+<br>```java
```
Comment (Aikovdp) [inline docs/contributing/structure-guide.md:179]:
```suggestion
```java
```
You can just remove that tag, it causes the code block to not be rendered correctly and wasn't there in the first place 

image

**Comment by DV8FromTheWorld** [comment]
Is there anything preventing approval on this?

**Review by DV8FromTheWorld — APPROVED** [review]
This looks fine to me from a content perspective, but I'm not an expert in this software to know if all of the changes are valid from a formatting perspective.

**Comment by Andre601** [comment]
> This looks fine to me from a content perspective, but I'm not an expert in this software to know if all of the changes are valid from a formatting perspective.

I tested this locally and it looked fine for me.

**Comment by Andre601** [comment]
> Would you be able to move `colors.css` to `stylesheets/` instead of `assets/css/`

I honestly forgot this and iirc didn't apply it yet?
Would do that first, so no merge until that first pls.

**Comment by Aikovdp** [comment]
> I tested this locally and it looked fine for me.

Yep, I checked it too and it all looks good to me. Once that last change has been applied we'll be able to merge

**Comment by Andre601** [comment]
Moved the css to `assets/stylesheets/`

The PR is now ready for me.

**Comment by Aikovdp** [comment]
Oh, I meant to move it to `stylesheets/`, not `assets/stylesheets`

**Comment by Andre601** [comment]
> Oh, I meant to move it to `stylesheets/`, not `assets/stylesheets`

That wouldn't make any sense... Javascript and CSS are usually stored under a `assets` directory in their respective folders.
Material for MkDocs also has `assets/stylesheets/` for its SCSS files, so it would make sense to keep it consistent here...

**Comment by Aikovdp** [comment]
> That wouldn't make any sense... Javascript and CSS are usually stored under a `assets` directory in their respective folders.

I get that, but like I said before, that's exactly what mkdocs shows in its own documentation: https://squidfunk.github.io/mkdocs-material/customization/#adding-assets
That makes it easier for contributors to figure out the structure, since we're not deviating from the norm. The more they can just get from the mkdocs material documentation the better, because in our contributing guidelines we'll need to state everything they can't know by reading those docs. 

> Material for MkDocs also has assets/stylesheets/ for its SCSS files, so it would make sense to keep it consistent here...

That's in the theme files themselves, not in their own documentation that's using the theme. We'd want to keep it consistent with the documentation, as we're doing the same.

**Comment by Andre601** [comment]
Made quite a few changes.

- Added a first design for a Landing Page. [[Preview](https://cdn.discordapp.com/attachments/919334419347677194/924103570343723118/unknown.png)]
  This isn't final yet and it's currently discussed and still worked on in the `Splash Page` thread
- Moved CSS from `assets/stylesheets/` to `stylesheets/`. Hope it's good that way.
- Changed the primary colour used in the wiki to use the actual colour used in the logo.
- Updated the logo. Got rid of the gradient and instead use a single colour.

Since the stuff isn't final yet do I convert this into a draft to prevent accidental merging.

File: docs/index.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+---
+template: overrides/home.html
```
[the remark this answers is not in the document]
Comment (Andre601) [reply docs/index.md:2]: No longer the case now since a first rough frontpage exist, which requires the file itself...

**Comment by Aikovdp** [comment]
Could you please handle the landing page and all that in a different PR? Those changes are entirely separate from everything else in here, and it's easier to review and merge those separately. We can get this PR merged already then

**Comment by Andre601** [comment]
> Could you please handle the landing page and all that in a different PR? Those changes are entirely separate from everything else in here, and it's easier to review and merge those separately. We can get this PR merged already then

Okay. I removed the override stuff, but I leave in the CSS stuff since it doesn't affect the page in any negative way

**Comment by DV8FromTheWorld** [comment]
Is there anything holding up this PR?

**Comment by Andre601** [comment]
I don't think so...

**Review by Aikovdp — APPROVED** [review]
As was discussed in the #wiki-dev channel on Discord, the previous two commits were reverted so that the splash page could be made in a different PR and this one can now be merged.

**Pull request merged**